src/flameGraph.cpp

```diff
@@ -479,7 +609,58 @@ double FlameGraph::printFrame(std::ostream& out, const std::string& name, const
     return framewidth;
 }
 
-int FlameGraph::selectFrameColor(std::string& name) {
+bool FlameGraph::sortMap(std::pair<std::string, Trie> a, std::pair<std::string, Trie> b)  {
+    return a.second._total > b.second._total; 
+}
+
+void FlameGraph::printTreeFrame(std::ostream& out, const std::string& name, const Trie& f, int type, int depth) {
+    double framewidth = f._total * _scale;
+    // Skip too narrow frames, they are not important
+    if (framewidth >= _minwidth) {
+        long childtotal = 0;
+        std::vector< std::pair<std::string, Trie> > pairs;
+        for (std::map<std::string, Trie>::const_iterator itr = f._children.begin(); itr != f._children.end(); ++itr)
+            pairs.push_back(*itr);
+
+        std::sort(pairs.begin(), pairs.end(), sortMap);
+
+        for (size_t i = 0; i < pairs.size(); ++i) { 
+            std::string full_title = pairs[i].first;
+            int color = selectFrameColor(full_title,false);
+            StringUtils::escape(full_title);
+            bool format = true;
+            if(pairs[i].second._children.size() == 0) {
+                format = false;
+            }
+            long childtotal = 0;
+            for (std::map<std::string, Trie>::const_iterator itr = pairs[i].second._children.begin(); itr != pairs[i].second._children.end(); ++itr)
+            {
+                childtotal +=  itr->second._total;
+            }
+	    if(type == CALL_TREE) {
+                snprintf(_buf, sizeof(_buf),
+                "<li><a href=\"#\">[%d] %.2f%% %lld self: %.2f%% %lld</a><span style=\"color: #%06x\"> %s</span>",
+                depth, pairs[i].second._total * _pct,pairs[i].second._total,(pairs[i].second._total-childtotal)* _pct,(pairs[i].second._total-childtotal),color,full_title.c_str()); 
+            }else {
+               snprintf(_buf, sizeof(_buf), 
+               "<li><a href=\"#\">[%d] %.2f%% %lld</a><span style=\"color: #%06x\"> %s</span>", depth, pairs[i].second._total * _pct,pairs[i].second._total,color,full_title.c_str());
```

Comment:
I'd suggest changing `style="color: xxx"` to `class="paletteName"`, so that
1) the output will be shorted;
2) colors will be customizable in css.
Color are better now, but still a little bit acid.

Reply:
Yes,  I am using class="paletteName" now.